core/config/toml/types.go

```diff
@@ -1434,14 +1439,71 @@ func (r *ExternalRegistry) setFrom(f *ExternalRegistry) {
 	}
 }
 
+type GatewayConnectorConfig struct {
+	NodeAddress               string
+	DonId                     string
+	Gateways                  []ConnectorGatewayConfig
+	WsClientConfig            network.WebSocketClientConfig
+	AuthMinChallengeLen       int
+	AuthTimestampToleranceSec uint32
+}
+
+func (r *GatewayConnectorConfig) setFrom(f *GatewayConnectorConfig) {
+	if f.NodeAddress != "" {
+		r.NodeAddress = f.NodeAddress
+	}
+
+	if f.DonId != "" {
+		r.DonId = f.DonId
+	}
+
+	// TODO: verify this copy by reference is ok, or does array need to be copied by value
+	if f.Gateways != nil {
+		r.Gateways = f.Gateways
+	}
+
+	if !reflect.ValueOf(f.WsClientConfig).IsZero() {
+		r.WsClientConfig = f.WsClientConfig
+	}
+
+	if f.AuthMinChallengeLen != 0 {
+		r.AuthMinChallengeLen = f.AuthMinChallengeLen
+	}
+
+	if f.AuthTimestampToleranceSec != 0 {
+		r.AuthTimestampToleranceSec = f.AuthTimestampToleranceSec
+	}
+}
+
+type ConnectorGatewayConfig struct {
```

Comment:
Note: you have both ConnectorGatewayConfig AND GatewayConnectorConfig which is a bit confusing here. Can we be more specific about what each config is for?

Reply:
I agree that it's confusing.  Note that the gateway/connector used to contain
```
type ConnectorConfig struct {
	NodeAddress               string
	DonId                     string
	Gateways                  []ConnectorGatewayConfig
	WsClientConfig            network.WebSocketClientConfig
	AuthMinChallengeLen       int
	AuthTimestampToleranceSec uint32
}

type ConnectorGatewayConfig struct {
	Id  string
	URL string
}
```

Reply:
And there's also a GatewayConfig outthere.

Reply:
We can rename this one into something more meaningful, maybe "GatewayNodeAddress" or "GatewayNodeIdentifier" (neither is perfect, though).